core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -92,6 +92,16 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
       }, 0, reviveIntervalMs, TimeUnit.MILLISECONDS)
     }
 
+    private def getSparkConfProperties(): Seq[(String, String)] = {
+      val properties = new ArrayBuffer[(String, String)]
+      for ((key, value) <- sparkConf.getAll) {
+        if (key.startsWith("spark.")) {
+          properties += ((key, value))
+        }
+      }
+      properties
+    }
```

Comment:
after you move it, I would add a comment that says:
```
// Retrieve fresh properties from SparkConf again in case a new property is set
// This is necessary for passing certain properties to executors in YARN (SPARK-8687)
```